demisto_sdk/commands/validate/validators/IF_validators/IF109_invalid_required_field.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import Iterable, List, Union, cast
+
+from demisto_sdk.commands.common.constants import GitStatuses
+from demisto_sdk.commands.content_graph.objects.incident_field import IncidentField
+from demisto_sdk.commands.content_graph.objects.incident_type import IncidentType
+from demisto_sdk.commands.content_graph.objects.indicator_field import IndicatorField
+from demisto_sdk.commands.content_graph.objects.indicator_type import IndicatorType
+from demisto_sdk.commands.validate.validators.base_validator import (
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Union[IncidentField, IndicatorField, IncidentType, IndicatorType]
+
+
+class IsValidRequiredFieldValidator(BaseValidator[ContentTypes]):
+    error_code = "IF109"
+    description = "Checks if the incident field required field is valid."
+    rationale = "In case an incident field is required, newly added associated incident types should be new."
+    # error_message = "{0}"
+    related_field = "required"
+    expected_git_statuses = [GitStatuses.MODIFIED, GitStatuses.ADDED]
+
+    def obtain_invalid_content_items(
+        self, content_items: Iterable[ContentTypes]
+    ) -> List[ValidationResult]:
+        types_items: list[str] = []
+        fields_items = []
+        for item in content_items:
+            if (
+                isinstance(item, IncidentType) or isinstance(item, IndicatorType)
+            ) and item.git_status == GitStatuses.ADDED:
+                types_items.append(item.object_id)
+            elif isinstance(item, IncidentField) or isinstance(item, IndicatorField):
+                fields_items.append(item)
+
+        return [
+            ValidationResult(
+                validator=self,
+                message=error_res,
+                content_object=content_item,
+            )
+            for content_item in fields_items
+            if (error_res := self.is_invalid_required_field(content_item, types_items))
+        ]
+
+    @staticmethod
+    def is_invalid_required_field(
+        content_item: Union[IncidentField, IndicatorField], added_types: list[str]
+    ):
+        # Required fields should not be associated to all
+        if content_item.required and content_item.associated_to_all:
+            return "Required field should not be associated to all types."
```

Comment:
Let's make the message clearer, A required {content_item_type} should not be associated with all types.